h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -431,7 +437,9 @@ public void compute2() {
             rs._tsk.join();
           } else if(rs.isMutating()) // c) => throw IAE
             throw new IllegalArgumentException("Can not compute rollup stats while vec is being modified. (3)");
-        } else { // d) => compute the rollups
+        } else { // rs == null d) => compute the rollups
+          final Vec vec = DKV.getGet(_vecKey);
+          if (vec == null) break; // Vec was deleted since we last checked
```

Comment:
This check is necessary to make my fix work.

The idea of my fix is to change the way Frames are deleted:
1. First delete the Vecs (and chunks) everywhere (all nodes).
2. Then delete the RollupStats.
(two independent MRTasks)

The old bulk_remove() deleted both RollupStats and Vec in the same MRTask. It was possible that RollupStats were deleted everywhere (and therefore null in the "else" branch) but Vec was still not completely removed and the check on 441 could return a valid Vec. This vec would be deleted in a little and leave an instance of RollupStats without a Vec.

This check is therefore necessary to make the fix work. If I don't make sure the Vec exists, rollupstats can be Calculated and inserted after the Vec is already deleted.